src/main/java/org/apache/sysml/lops/FunctionCallCPSingle.java

```diff
@@ -0,0 +1,129 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysml.lops;
+
+
+import java.util.ArrayList;
+
+import org.apache.sysml.hops.Hop;
+import org.apache.sysml.hops.HopsException;
+import org.apache.sysml.lops.LopProperties.ExecLocation;
+import org.apache.sysml.lops.LopProperties.ExecType;
+import org.apache.sysml.lops.compile.JobType;
+import org.apache.sysml.parser.DMLProgram;
+import org.apache.sysml.parser.Expression.DataType;
+import org.apache.sysml.parser.Expression.ValueType;
+
+public class FunctionCallCPSingle extends Lop  
+{	
+	private String _fnamespace;
+	private String _fname;
+	private String[] _output;
+	private ArrayList<Lop> _outputLops = null;
+
+       public FunctionCallCPSingle(ArrayList<Lop> inputs, String fnamespace, String fname, String[] output, ArrayList<Hop> outputHops, ExecType et) 
+		throws HopsException, LopsException 
+	{
+		this(inputs, fnamespace, fname, output, et);
+		if(outputHops != null) {
+			_outputLops = new ArrayList<Lop>();
+			for(Hop h : outputHops)
+				_outputLops.add( h.constructLops() );
+		}
+	} 
+	
+       public FunctionCallCPSingle(ArrayList<Lop> inputs, String fnamespace, String fname, String[] output, ExecType et) 
+	{
+		super(Lop.Type.FunctionCallCP, DataType.UNKNOWN, ValueType.UNKNOWN);	
+		//note: data scalar in order to prevent generation of redundant createvar, rmvar
+		
+		_fnamespace = fnamespace;
+		_fname = fname;
+		_output = output;
+		
+		//wire inputs
+		for( Lop in : inputs ) {
+			addInput( in );
+			in.addOutput( this );
+		}
+			
+		//lop properties: always in CP
+		boolean breaksAlignment = false; 
+		boolean aligner = false;
+		boolean definesMRJob = false;
+		lps.addCompatibility(JobType.INVALID);
+		lps.setProperties(inputs, et, ExecLocation.ControlProgram, breaksAlignment, aligner, definesMRJob );
+	}
+
+	public ArrayList<Lop> getFunctionOutputs() {
+		return _outputLops;
+	}
+	
+	@Override
+	public String toString() {
+		return "function call: " + DMLProgram.constructFunctionKey(_fnamespace, _fname);
+	}
+	
+	/**
+	 * Method to generate instructions for external functions with multiple returns.
+	 */
+	@Override
+	public String getInstructions(String[] inputs, String[] output) throws LopsException
+	{		
```

Comment:
please override the following method instead:
```
	@Override
	public String getInstructions(String[] inputs, String output) throws LopsException {
		return getInstructions(inputs, new String[]{output});
	}
```

Instead of compiling an array of output names into the instruction, you simply take the passed output parameter as the single output. During instruction generation, this method will be called according to the number of inputs. Please have a look at the existing lop `SpoofFused` how to do that in the general case.